x/sqlite.ts

```diff
@@ -0,0 +1,105 @@
+import * as sqlite from "https://deno.land/x/sqlite@v2.3.2/mod.ts";
+
+import * as driver from "../sql/driver.ts";
+
+export type Value = null | boolean | number | string | bigint | Uint8Array;
+
+export type Meta = driver.Meta<{
+  Value: Value;
+}>;
+
+export class Driver
+  implements driver.Driver<Meta>, driver.ConnectorOpener<Meta> {
+  openConnectorSync(path: string) {
+    return new Connector(this, path);
+  }
+}
+
+export class Connector implements driver.Connector<Meta> {
+  constructor(
+    readonly driver: Driver,
+    private readonly path: string,
+  ) {}
+
+  connectSync() {
+    const handle = new sqlite.DB(this.path);
+    return new Connection(this.driver, handle);
+  }
+}
+
+export class Connection implements driver.Connection<Meta> {
+  private activeTransaction?: Transaction;
+
+  constructor(
+    readonly driver: Driver,
+    private readonly handle: sqlite.DB,
+  ) {}
+
+  startTransactionSync() {
+    const transaction = new Transaction(this.driver, this.handle, this);
+    this.handle.query("BEGIN DEFERRED TRANSACTION").return();
+    return transaction;
+  }
+
+  lastInsertedIdSync(): Value {
+    return this.handle.lastInsertRowId;
+  }
+
+  affectedRowsSync(): number {
+    return this.handle.changes;
+  }
+
+  closeSync() {
+    this.handle.close();
+  }
+}
+
+export class Transaction implements driver.Transaction<Meta> {
+  private childTransaction?: Transaction;
+
+  constructor(
+    readonly driver: Driver,
+    private readonly connectionHandle: sqlite.DB,
+    parent: Transaction | Connection,
+  ) {
+    if (parent instanceof Transaction) {
```

Comment:
SQLite transactions do not nest, I think you want to use safe points for nested transactions: https://www.sqlite.org/lang_transaction.html

Reply:
You're right, thanks. I'm going to use `transaction` for the outermost because it (hypothetically) allows the flexibility of creating `immediate` transactions, but then `savepoint`s inside that.

Reply:
Updated the in-progress driver code to reflect this intended logic.
If we're in a Connection, we
```ts
    this.handle.query("BEGIN DEFERRED TRANSACTION").return();
```
if we're already in a transaction, we
```ts
    this.handle.query(`SAVEPOINT TRANSACTION_${this.depth}`).return();
```
where `this.depth` is how deeply nested the transaction is. (I think this is adequate, because sqlite doesn't allow multiple transactions at the same level in the same connection, each transaction has one parent and zero or one children, if I understand correctly).

When committing a transaction:
```ts
    if (this.parent instanceof Connection) {
      this.handle.query("COMMIT TRANSACTION").return();
    } else {
      this.handle.query(`RELEASE TRANSACTION_${this.depth}`).return();
    }
```

When rolling back:
```ts
    if (this.parent instanceof Connection) {
      this.handle.query("ROLLBACK TRANSACTION").return();
    } else {
      this.handle.query(`ROLLBACK TO TRANSACTION_${this.depth}`).return();
    }
```

Let me know if anything about this approach sounds confused (the code isn't complete enough to actually run it yet).